### PR TITLE
[processor/resourcedetection][exporter/datadog] Supress unwanted warning logs coming from EC2 metdata provider

### PR DESCRIPTION
**Pull request by matej-g**
**Description:** 
Some users are seeing unwanted logs that are coming from the `metadaproviders` package, specifically from EC2 provider. This is caused by the fact that the portion of the code in AWS SDK does not respect the default logger configuration (logging is turned off by default for the SDK). To sidestep the issue, the suggested solution in this PR is to introduce simple custom logging function to supress any output from SDK.

**Link to tracking Issue:** #22807

**Testing:** Manually tested

**Documentation:** N/A

### CODE REVIEW

**Review by bryan-aguilar — COMMENTED** [review]
I'm a little wary of suppressing all SDK logs. What is the impact from this bug, how noisy are these logs? 

Edit: I see I misread the original issues, is logging off by default? Is there documentation you can link to so we can verify that?

File: internal/metadataproviders/aws/ec2/metadata.go [hunk 1]
```diff
@@ -22,9 +23,14 @@ type metadataClient struct {
 
 var _ Provider = (*metadataClient)(nil)
 
+// New Provider returns a new AWS EC2 metadata provider. The provider is established
+// with a custom logger to suppress any SDK logs, in order to work around an issue in the AWS SDK Go
+// (https://github.com/aws/aws-sdk-go/issues/5116), which causes unwanted output to appear on
+// the stdout. This change can be reverted once the issue is resolved upstream and a new version
```
Comment (songy23) [inline internal/metadataproviders/aws/ec2/metadata.go:29]: Can we add a tracking issue for this?

**Comment by matej-g** [comment]
> I'm a little wary of suppressing all SDK logs. What is the impact from this bug, how noisy are these logs?
> 
> Edit: I see I misread the original issues, is logging off by default? Is there documentation you can link to so we can verify that?

- You can take a look at `LogLevelType` documentation https://pkg.go.dev/github.com/aws/aws-sdk-go/aws#LogLevelType
- It's being used as a default value here https://github.com/aws/aws-sdk-go/blob/v1.49.6/aws/defaults/defaults.go#L63

File: internal/metadataproviders/aws/ec2/metadata.go (hunk 1 above)
Comment (matej-g) [reply internal/metadataproviders/aws/ec2/metadata.go:29]: Hope this suffices https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30139

**Review by songy23 — APPROVED** [review]
LGTM, pinging code owners @dashpole @Aneurysm9

**Review by bryan-aguilar — APPROVED** [review]

**Comment by matej-g** [comment]
This is already obsolete as it has been fixed in the upstream (https://github.com/aws/aws-sdk-go/releases/tag/v1.49.16) and will be fixed by merging the PR to update AWS dependencies - https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/30341. Closing this.

**Pull request closed without merging**